text/xxxx-metrics.md

```diff
@@ -0,0 +1,318 @@
+release it's possible to narrow down the transactions to look at in the performance section
+of Sentry.  Likewise by tagging metrics with releases it's possible to directly drive the
+release health system.  When a measurement is emitted, directly correlation is possible in
+that percentiles can be calculated from metrics timeseries data, and then the p95 or similar
+can be taken as the lower bound for finding transactions or spans where that number is the
+lower bound of that measurement.
+
+```python
+from sentry_sdk import metrics
+
+# When the metrics is thrown into the aggregator, it's enhanced by the tags
+# `release`, `environment` and `transaction` from the current trace.
+with metrics.timing("my-expensive-operation", tags={"kind": "batch"}):
+    # process the batch here.
+    ...
+```
+
+## Units and Convertability
+
+When metrics are emitted, units are sent alongside.  This means that the meaning of a number
+is obvious and re-calculations are possible in the UI.  SDKs are expected to have reasonable
+unit defaults for timings they are automatically emitting.  Units that are not understood by
+the system are called "custom" units.  Units are always in singular form.
+
+```python
+metrics.distribution("upload.file_size", attachment.bytes, unit="byte")
+metrics.gauge("cpu.pointer_size", CPU_POINTER_SIZE, unit="bit")
+metrics.incr("processor.messages_processed", 1, unit="message")
+```
+
+## MRIs
+
+Metrics are identified by their [metrics resource identifier](https://getsentry.github.io/relay/relay_metrics/struct.MetricResourceIdentifier.html).
+It consists of type, namespace, metric name and unit.  On the SDK side in both
+memory and in the transport the MRI does not play much of a role as it's encoded
+in a more relaxed format.  Most importantly the namespace is implied as "custom"
+for custom metrics.  The understanding of MRIs however is important as in the final
+storage the metric is keyed by it's MRI in the system which means that the following
+metrics are all different:
+
+```
+d:custom/endpoint.response_time@millisecond
+d:custom/endpoint.response_time@second
+g:custom/endpoint.response_time@second
+```
+
+For some metrics the idea is that some unification can take place later.  For
+instance when an SDK sends seconds and another SDK version sends milliseconds
+for the same unit, the system should later be able to unify them at query or
+ingestion time.  This type of handling would not extend to custom or
+incompatible types.
+
+## Span Accounting
+
+The plan is that spans in the `custom/` namespace are generally billed for in
+one way or another.  Other metrics would be billed as part of other product
+features.
+
+# SDK Design
+
+SDKs are expected to provide basic metric emission APIs modelled after common APIs inspired
+by statsd.
+
+## Aggregations
+
+SDKs are expected to aggregate locally for windows of up to 10 seconds.  The aggregator shall be
```

Comment:
I guess that mobile and browser SDKs can apply a different strategy cause the metrics frequency will be lower. One strategy could be to attach these buckets to an already scheduled envelope, which we do for [client reports](https://develop.sentry.dev/sdk/client-reports/).